lineapy/plugins/serializers/mlflow_io.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import sys
+from inspect import getmodule
+from typing import Any, Optional
+
+from lineapy.data.types import MLflowArtifactInfo
+from lineapy.utils.config import options
+from lineapy.utils.logging_config import configure_logging
+
+"""
+Serializer and deserializer for MLflow supported model flavors. Only include 
+annotated modules at this time. 
+
+[x] sklearn
+[x] prophet
+[ ] tensorflow (need update current annotation)
+[ ] keras (need update current annotation)
+[ ] h2o
+[ ] gluon
+[x] xgboost
+[ ] lightgbm
+[ ] catboost
+[ ] spacy
+[ ] fastai
+[x] statsmodels
+
+"""
+
+
+logger = logging.getLogger(__name__)
+configure_logging()
+
+mlflow_io = {}
+
+try:
+    import mlflow
+
+    try:
+        from sklearn.base import BaseEstimator
+
+        mlflow_io["sklearn"] = {
+            "class": [BaseEstimator],
+            "serializer": mlflow.sklearn.log_model,
+            "deserializer": mlflow.sklearn.load_model,
+        }
+    except Exception:
+        pass
+
+    try:
+        from prophet import Prophet
+
+        mlflow_io["prophet"] = {
+            "class": [Prophet],
+            "serializer": mlflow.prophet.log_model,
+            "deserializer": mlflow.prophet.load_model,
+        }
+    except Exception:
+        pass
+
+    try:
+        from xgboost.core import Booster
+
+        mlflow_io["xgboost"] = {
+            "class": [Booster],
+            "serializer": mlflow.xgboost.log_model,
+            "deserializer": mlflow.xgboost.load_model,
+        }
+    except Exception:
+        pass
+
+    try:
+        from statsmodels.base.wrapper import ResultsWrapper
+
+        mlflow_io["statsmodels"] = {
+            "class": [ResultsWrapper],
+            "serializer": mlflow.statsmodels.log_model,
+            "deserializer": mlflow.statsmodels.load_model,
+        }
+    except Exception:
+        pass
+
+except ImportError:
+    pass
+
+
+def try_write_to_mlflow(value: Any, name: str, **kwargs) -> Optional[Any]:
+    """
+    Try to save artifact with MLflow
+
+    Parameters
+    ----------
+    value: Any
+        value(ML model) to save with mlflow
+    name: str
+        artifact_path and registered_model_name used in
+        `mlflow.sklearn.log_model` or equivalent flavors
+    **kwargs:
+        args to pass into `mlflow.sklearn.log_model` or equivalent flavors
+
+    Returns
+    -------
+    Optional[Any]
+        return a ModelInfo(MLflow model metadata) if successfully save with
+        mlflow; otherwise None.
+
+    Note that, using Any for type checking in case mlflow is not installed.
```

Comment:
Move this comment up to the top of the docstring. This is actually incredibly important for people to understand the "failure mode" for this function.